prim_ops.c

```diff
@@ -1136,7 +1136,7 @@ u8 rol_byte(u8 d, u8 s)
 		CONDITIONAL_SET_FLAG(s == 1 &&
 							 XOR2((res & 0x1) + ((res >> 6) & 0x2)),
 							 F_OF);
-	} if (s != 0) {
+	} else if (s != 0) {
```

Comment:
adding the 'else' is actually a logical no-op, but the overflow flag handling is wrong; for this, I've rewritten
the 6 rol/ror functions and added tests for them; pr follows